src/jsonpath/lexer.c

```diff
@@ -260,18 +260,25 @@ lex_token scan(char** p, char* buffer, size_t bufSize, lex_error* err)
 	return found_token;
 }
 
-/* Extract contents of string bounded by single or double quotes */
+/* Extract contents of string bounded by either single or double quotes */
 static bool extract_quoted_literal(char* p, char* buffer, size_t bufSize, lex_error* err)
 {
-
 	char* start;
+	char quote_type;
```

Comment:
@crocodele I could be wrong, but I _think_ `quote_type` is instantiated with whatever garbage exists at the memory location. The garbage could be a quote, which would cause some unexpected behavior. Maybe play it safe and initialize it to something inert like `0`?

Reply:
Hmm there might not be a safe default value for `quote_type` that would be guaranteed to not cause the loop at 281 to short circuit. Maybe introduce a `bool quote_found` variable?

Reply:
Ideally the whole function will only ever be called when there is a quote to be found, but better safe than sorry. :slightly_smiling_face: 

Added `bool quote_found`: https://github.com/supermetrics/php-ext-jsonpath/pull/23/commits/fc7d3b03109c665d40d03d061d07c3e8c22612dc

Reply:
Ah yes, good point.